src/time_bucket.c

```diff
@@ -234,6 +234,25 @@ ts_timestamp_bucket(PG_FUNCTION_ARGS)
 	}
 }
 
+TS_FUNCTION_INFO_V1(ts_timestamp_offset_bucket);
+
+TSDLLEXPORT Datum
+ts_timestamp_offset_bucket(PG_FUNCTION_ARGS)
+{
+	Datum period = PG_GETARG_DATUM(0);
+	Datum timestamp = PG_GETARG_DATUM(1);
+
+	if (TIMESTAMP_NOT_FINITE(DatumGetTimestamp(timestamp)))
```

Comment:
Could you add a comment why this test is needed? And is it possible to construct a test case for this?

Reply:
We have the similar code in ts_timestamp_bucket and relevant functions , and there are tests for these cases in timestamp.sql

Reply:
I was asking for a test case because when I did the review, the return in Line 246 was not covered by a test case. Currently, there are failing tests for this PR, and the recent test coverage is not reported. Could you look at the test coverage when the tests have been run successfully? If the current tests do not cover the return, please add an appropriate test case.

Reply:
Okay

Reply:
I have added the test cases